lua/nvim-tree/view.lua

```diff
@@ -473,7 +473,7 @@ end
 function M.get_winnr(tabpage)
   tabpage = tabpage or vim.api.nvim_get_current_tabpage()
   local tabinfo = M.View.tabpages[tabpage]
-  if tabinfo ~= nil then
+  if tabinfo ~= nil and vim.api.nvim_win_is_valid(tabinfo.winnr) then
```

Comment:
@alex-courtis I'm hunting down a bug that seems to be related to this change. Can you help me out in figuring out what is the problem? 🙏

What I currently know is that when I call:

```
api.node.open.replace_tree_buffer
```

I get the following error:

```
E5108: Error executing lua: ...config/nvim/plugged/nvim-tree.lua/lua/nvim-tree/view.lua:476: Expected Lua number
stack traceback:
        [C]: in function 'nvim_win_is_valid'
        ...config/nvim/plugged/nvim-tree.lua/lua/nvim-tree/view.lua:476: in function 'get_winnr'
        ....config/nvim/plugged/nvim-tree.lua/lua/nvim-tree/lib.lua:22: in function 'get_node_at_cursor'
        ....config/nvim/plugged/nvim-tree.lua/lua/nvim-tree/api.lua:61: in function <....config/nvim/plugged/nvim-tree.lua/lua/nvim-tree/api.lua:60>
```

Do you maybe have an idea where should I look next?

Reply:
This change did indeed surface some issues, many thanks for digging into this!

Can you log the value of tabinfo.winnr? As of current master it's nil checked, however anything could be in there.

We could just do a `type(tabinfo.winnr) == "number"`...

#2628 will do away with a lot of these issues.